src/game_api/entity_db.cpp

```diff
@@ -97,7 +97,7 @@ EntityDB* get_type(uint32_t id)
     // Special case: map_ptr might be 0 if it's not initialized.
     // This only occurs in list_entities; for others, do not check the pointer
     // to see if this assumption works.
-    if (!entity_factory_ptr)
+    if (!entity_factory_ptr || id > 0x393)
```

Comment:
We always tented to use `to_id` instead of hard coded type id, but as we don't see any updates in the future it's probably fine for hardcoded values now

Reply:
Yeah the actual array size seemed hardcoded anyway and even if a new entity was added, we woudn't know what it was and couldn't use to_id to get the last id.

Reply:
well, not often they add new entity at the end (as they did in last big update xd)
btw. this can still crash as there are gaps in the id's, so it's not even size, it's just last id